### PR TITLE
the great gaming rework

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

adds dynamic gamemode
and aims to rework almost all major gamemodes in favor of round continuability (i came up with that word on the spot)
also renames passenger to assistant :trollface: 

### Dynamic gamemode
Generates a random amount of points that will be used for generating roundstart and midround threats.
Antag related events will be handled by the dynamic gamemode if it's present.
Also in dynamic zombies are midround only.

### Revolutionaries changes
- Heads of staff now cannot be revved. Cuffing or exiling them will still result in revolutionaries' win. ~~(TODO)~~
- When all heads of staff get handled, all & future mindshielded people will get the Enemy of the Revolution role that will make them able to see rev icons (headrev will be shown as a regular revolutionary). The gamemode will truly end when all Enemies of the Revolution are handled, like heads of staff. ~~(TODO)~~
- Evac may not be called while both sides are alive & on the station and there's at least 35% of crew converted or EotRs left. ~~(TODO)~~

### Nukie changes
- Nukies now get random syndicate reinforcement agent names, so expect seeing "John Kills-a-lot" instead of "Operator India" :trollface: ~~(TODO)~~
- Evac may not be called while nukies are present on the station. It may be called when all of the nukeops are either far away in space or dead, shortly, not on the station :trollface: ~~(TODO)~~
- Lone operative will now spawn on his own nukie planet and is treated like a regular nukie, but with 300 TC and a war declarator (challenge mode). (TODO)

### Tritar changes
- Traitors gained factions. From ARC terrorist to MI8 agent and corporate climber. ~~(TODO)~~
- Increased difficulty to give them more objectives. ~~(TODO)~~

### Changelinges (changeling changes, get it? get it? get it?..)
- Upped biomass cap from 30 to a whopping 60 expecting less murderbonering and actual stealth play from clings. ~~(TODO)~~
They are still free agents and can do whatever they want.

### Miscellaneous changes
- Changed passenger role to Asisstant, to keep up with tradition :trollface:
- Resprited syndicate elite hardsuit helmet to make it more syndicate-like.

## Why / Balance
making gamemodes more fun and complicated, forcing crew to play by the rules

### Why do this in a big single PR if you can divide it into multiple smaller ones
i have a shit ton of ideas and i don't want to jump from branch to branch

## Media
early dynamic prototype screenshot
image

traitor variety
image

:cl:
- add: Added dynamic gamemode.
- tweak: Nuclear operatives now spawn with random syndicate reinforcement agent names.
- tweak: Lone Operative now spawns on nukie planet.
- add: Added traitor variety.
- tweak: Now certain gamemodes will not end the round instantly (nukies/revs).
- tweak: Changed passenger role to assistant.

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive dynamic game rules for both roundstart and mid-round events, enhancing gameplay variety.
	- Added new antagonist roles and spawning mechanisms, including dynamic mid-round traitors, changelings, and unique events like zombie outbreaks.
	- Implemented localization strings for dynamic game mechanics and threat levels, providing players with essential information.
	- Updated game rules for Nuke Operatives, improving shuttle call handling and round end behavior.
	- Introduced a new component for managing dynamic station events and scheduling.
	- Enhanced threat management systems with new budgets and dynamic rule components.

- **Bug Fixes**
	- Updated naming conventions for ghost role rules for consistency and clarity.
	- Adjusted test logic to reflect changes in round termination behavior related to Nuke Operatives.

- **Documentation**
	- Enhanced comments and annotations within game rules to improve understanding of modifications and functionalities.

### CODE REVIEW

**Comment by BombasterDS** [comment]
Revs call shuttle, not end round instanly now

**Comment by BombasterDS** [comment]
Still don't understand how this will work. It will spawn waves of antag during the round depending on how much players in the round?

**Comment by GroggleG** [comment]
> Still don't understand how this will work. It will spawn waves of antag during the round depending on how much players in the round?

The original dynamic system is like a obese (but poor) chef buying food. They bring a specific amount of cash each visit to the market and buy based on a few different parameters. They always buy a main course with a lot of room to add side dishes.

**Comment by GroggleG** [comment]
https://tgstation13.org/wiki/Game_Mode#Dynamic
Or you can read this IDK
*picks nose

**Comment by Aidenkrz** [comment]
Someone already made this on wizden, and it's pretty good.

**Comment by whateverusername0** [comment]
@Piras314 any comments?

**Comment by Aineias1** [comment]
this is gonna be fun, ping me when its merged

**Comment by BombasterDS** [comment]
Honestly, i don't like revolution changes. Enemies of revolution may be cool in case you killed all command members (if revs killed all command members it means they want to fight with sec too), but this will only lead to more murderboning. We already have lings that needs to murderbone to survive, heretics that needs murderbone to progress and nukies that have mission to detonate nuclear bomb. Make all antags murderboning is not cool.

**Comment by whateverusername0** [comment]
> Honestly, i don't like revolution changes. Enemies of revolution may be cool in case revs killed all command members (if revs killed all command members it means they want to fight with sec too), but this will only lead to more murderboning. We already have lings that needs murderbone to survive, heretics that needs murderbone to progress and nukies that have mission to detonate nuclear bomb. Make all antags murderboning is not cool.

Evac is gonna be called automatically after assigning enemies of the revolution "to gather all left out loyal personnel", like with zombies, so ghosts won't have to wait that long.

Also gonna update clings to not be that murderboney

**Comment by whateverusername0** [comment]
I like the TG formula that translates to "Crew's here to bring good, Antags are here to bring bad".

**Comment by BombasterDS** [comment]
> I like the TG formula that translates to "Crew's here to bring good, Antags are here to bring bad". Restricting someone to their objectives is a boring way to deal with massive amounts of murderbone. Crew just has to git gud and gain some protag energy to deal with these feral antags

Formula don't work. Tiders still main problem to station.

File: Content.Server/Antag/Components/AntagSelectionComponent.cs [hunk 1]
```diff
@@ -11,7 +11,10 @@
 
 namespace Content.Server.Antag.Components;
 
-[RegisterComponent, Access(typeof(AntagSelectionSystem), typeof(AdminVerbSystem))]
+// goob edit - no more specified access.
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Server/Antag/Components/AntagSelectionComponent.cs:14]: i am going full :godmode: mode
less restrictions = more control

[the remark this answers is not in the document]
Comment (Piras314) [reply Content.Server/Antag/Components/AntagSelectionComponent.cs:14]: :goberserk:

**Comment by whateverusername0** [comment]
@Piras314 don't merge just yet

**Comment by whateverusername0** [comment]
fuck

File: Resources/Prototypes/_Goobstation/GameRules/Dynamic/midround.yml [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  - type: StationEvent
+    earliestStart: 20
+    weight: 3
+    minimumPlayers: 15
+    startAnnouncement: station-event-communication-interception
+    startAudio:
+      path: /Audio/Announcements/intercept.ogg
+    duration: null
+    occursDuringRoundEnd: false
+  - type: AlertLevelInterceptionRule
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Resources/Prototypes/_Goobstation/GameRules/Dynamic/midround.yml:42]: check the changeling component

**Comment by whateverusername0** [comment]
image
dogshit test fail

**Review by Piras314 — CHANGES_REQUESTED** [review]
peak tbh but put antag command in a different pr. other than that the code and everything else looks good

**Comment by whateverusername0** [comment]
>  put antag command in a different pr.

damn you saw through this :finnadie:

**Review by Piras314 — APPROVED** [review]

**Comment by Piras314** [comment]
checkfail seems to be related

**Comment by whateverusername0** [comment]
image
checkfail says absolutely nothing about the error.

**Comment by whateverusername0** [comment]
@Piras314 now this is bs
image

**Comment by Piras314** [comment]
seems to happen every run, so you gotta fix it @whateverusername0

**Comment by a-person5660** [comment]
>Evac may not be called while nukies are present on the station. It may be called when all of the nukeops are either far away in space or dead, shortly, not on the station :trollface:

make sure this only applies for warops else it will give away that the shift is nukies.

**Comment by whateverusername0** [comment]
this test fail is unrelated to dynamic whatsoever wtf

**Comment by whateverusername0** [comment]
@Piras314 i done it

**Comment by JoeHammad1844** [comment]
I think this is mostly sufficient to close #372
thoughts?

**Pull request merged**